### DISCUSSION TITLE
Q&A — Phase 35.3 QuantumFeatureMap

### DISCUSSION THREAD

**Discussion by web3guru888** [web3guru888/asi-build#731] [Q&A]
### Q1: What are the tradeoffs between amplitude encoding and angle encoding?

**A:** The key tradeoffs are:

| Aspect | Amplitude Encoding | Angle Encoding |
|--------|-------------------|----------------|
| Qubits needed | O(log d) | O(d) |
| Circuit depth | O(2ⁿ) general, O(poly n) structured | O(1) per feature |
| State preparation | Complex, requires specific circuits | Simple rotation gates |
| Gradient computation | Difficult (no parameter-shift) | Easy (parameter-shift rule) |
| Expressibility | High (full Hilbert space) | Moderate (rotational subspace) |
| Hardware feasibility | Challenging for NISQ | NISQ-friendly |

**Recommendation**: Use angle encoding for NISQ devices and datasets with d ≤ 20. Use amplitude encoding when d is large and efficient state preparation circuits are known (e.g., for data with tree structure or sparse representations).

### Q2: How does kernel alignment help in feature map selection?

**A:** Kernel alignment measures the similarity between the quantum kernel matrix K_Q and the ideal kernel K_ideal = y·yᵀ (outer product of labels):

```
A(K_Q, K_ideal) = ⟨K_Q, K_ideal⟩_F / (||K_Q||_F · ||K_ideal||_F)
```

Values range from -1 to 1, with higher alignment indicating the kernel separates classes well. In practice:
1. Compute K_Q for candidate feature maps on a small calibration set
2. Evaluate alignment score for each
3. Select the feature map with highest alignment
4. Optionally, jointly optimize feature map parameters to maximize alignment (trainable kernels)

This avoids expensive cross-validation and gives a principled selection criterion grounded in learning theory (Cristianini et al. 2001).

### Q3: How does feature map depth affect classification performance?

**A:** Deeper feature maps (more layers of encoding + entangling gates) increase expressibility but introduce three costs:
1. **Gate errors accumulate**: Each additional layer adds ~0.5-2% error on NISQ hardware, reducing kernel estimation accuracy.
2. **Barren plateau risk**: Deep encoding circuits may produce nearly Haar-random states, making all kernel entries approach 1/2ⁿ (exponentially small), rendering the kernel matrix close to identity.
3. **Training overhead**: More parameters in trainable feature maps increase optimization cost.

Empirically, 2–4 layers of entangling+encoding blocks perform best for most datasets on 5–15 qubit systems. We provide an automatic depth selection via cross-validated kernel alignment.

### Q4: How does classical shadow estimation work for kernel computation?

**A:** Classical shadows (Huang et al. 2020) are a measurement-efficient protocol:
1. Prepare quantum state |φ(x)⟩
2. Apply a random Clifford unitary U (from a group — Pauli or Clifford)
3. Measure in the computational basis, obtaining outcome b
4. Store the classical snapshot: ρ̂ = M⁻¹(U† |b⟩⟨b| U) where M is the measurement channel inverse
5. Average over T random snapshots to estimate any observable

For kernel estimation, ⟨φ(x_i)|O|φ(x_j)⟩ can be estimated from the shadows of both states. The key advantage: T = O(log N) snapshots suffice to estimate N kernel entries simultaneously, reducing the O(N²) measurement cost to O(N log N). The tradeoff is that shadow estimation has higher variance per entry, requiring careful error analysis.

### Q5: What is the data re-uploading strategy and when should it be used?

**A:** Data re-uploading (Pérez-Salinas et al. 2020) encodes the same data point x multiple times across L circuit layers, each with different trainable parameters:

```
|ψ(x)⟩ = U_L(x, θ_L) ··· U_2(x, θ_2) · U_1(x, θ_1) |0⟩
```

This is analogous to a neural network where each layer re-processes the input. Key properties:
- **Universal approximation**: A single qubit with L layers can approximate any function f: [0,1]ᵈ → [-1,1] to arbitrary precision (for sufficient L)
- **Qubit-efficient**: Only O(1) qubits needed for arbitrary dimensional data
- **Trainable**: Parameters θ_l are optimized via gradient descent
- **Best for**: Low-qubit scenarios, function fitting, when the quantum advantage claim is not the priority but quantum-native feature extraction is desired

### Q6: When should kernel methods be preferred over variational classifiers?

**A:** The kernel approach (compute K, feed to classical SVM) and the variational approach (train parameterized circuit end-to-end) have different strengths:

**Prefer kernel methods when:**
- Training set is small (N < 1000) — kernel SVM has strong guarantees with limited data
- You want interpretability — kernel alignment and support vectors are well-understood
- No barren plateau risk — the feature map is fixed, not optimized
- Rigorous generalization bounds are needed (Rademacher complexity)

**Prefer variational classifiers when:**
- Large training sets — kernel matrix computation is O(N²) which becomes expensive
- The feature map should adapt to the data — jointly optimizing encoding and classification
- Shot budget is limited — variational circuits need O(1) measurements per training step vs O(N²) for full kernel

### Q7: What inductive biases do quantum feature maps introduce?

**A:** Quantum feature maps impose specific structural biases:
1. **Periodicity**: Rotation encoding with Ry(x) is 2π-periodic, biasing toward periodic functions. Mitigation: use non-linear pre-processing (e.g., arcsin scaling).
2. **Entanglement structure**: The connectivity of entangling gates determines which feature interactions are captured. Linear entanglement captures nearest-neighbor interactions; all-to-all captures arbitrary pairwise correlations.
3. **Symmetry**: Feature maps can be designed to respect known data symmetries (permutation invariance via symmetric ansätze, translational invariance via convolutional quantum circuits).
4. **Hilbert space geometry**: The quantum state space has a natural Fubini-Study metric, which defines the notion of similarity between data points. This is different from Euclidean distance and may or may not match the problem structure.

Understanding these biases helps select or design feature maps appropriate for specific data domains.
